cmd/frontend/debug.go

```diff
@@ -0,0 +1,43 @@
+//go:build debug
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os/exec"
+	"os/signal"
+	"syscall"
+)
+
+func waitForDebug(ctx context.Context) error {
+	pid := fmt.Sprintf("%d", syscall.Getpid())
+	cmd := exec.Command(
+		"dlv",
+		"attach",
+		"--api-version=2",
+		"--headless",
+		"--listen=unix:/dlv.sock",
+		"--allow-non-terminal-interactive",
+		pid,
+	)
+	cmd.Stdout = io.Discard
+	cmd.Stderr = io.Discard
+
+	if err := cmd.Start(); err != nil {
+		return err
+	}
+
+	go func() {
+		if err := cmd.Wait(); err != nil {
+			panic(err)
+		}
+	}()
+
+	s, cancel := signal.NotifyContext(ctx, syscall.SIGCONT)
```

Comment:
I wonder if instead of using a signal here, if we can look at `/proc/self/status`: https://man7.org/linux/man-pages/man5/proc_pid_status.5.html

There should be a field called `TracerPid` which would be non-zero when being traced.

*IF* this works, there'd be no need to interact with the process at all other than to have dlv attach to it.

Reply:
This works!

Out of curiosity, how did you know this / find it out?